panic: runtime error: slice bounds out of range [:4] with length 0

When one create a MR passing `--cover-letter` the text editor is opened to allow the one to write a description message.
When the file is saved and closed no content it prints the message: `2021/02/11 10:32:35 mr_create.go:243: aborting MR due to empty MR msg`, which is fine and correct.

However, if the user also passes `--draft` and save&close and empty file:
```
panic: runtime error: slice bounds out of range [:4] with length 0

goroutine 1 [running]:
github.com/zaquestion/lab/cmd.runMRCreate(0x18c9a60, 0xc0005f6600, 0x0, 0x3)
        /home/bmeneg/git/lab/cmd/mr_create.go:203 +0x1da5
github.com/spf13/cobra.(*Command).execute(0x18c9a60, 0xc0005f65d0, 0x3, 0x3, 0x18c9a60, 0xc0005f65d0)
        /home/bmeneg/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:854 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x18caf60, 0xc000032240, 0x6, 0x6)
        /home/bmeneg/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /home/bmeneg/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/zaquestion/lab/cmd.Execute()
        /home/bmeneg/git/lab/cmd/root.go:241 +0x188
main.main()
        /home/bmeneg/git/lab/main.go:28 +0x93
```

I'm going to take a look at that later.

--- a/cmd/mr_create.go
+++ b/cmd/mr_create.go
@@ -229,10 +229,10 @@ func runMRCreate(cmd *cobra.Command, args []string) {
 
 	draft, _ := cmd.Flags().GetBool("draft")
 	if draft {
-		isWIP := strings.EqualFold(title[0:4], "wip:")
-		isDraft := strings.EqualFold(title[0:6], "draft:") ||
-			strings.EqualFold(title[0:7], "[draft]") ||
-			strings.EqualFold(title[0:7], "(draft)")
+		isWIP := hasPrefix(title, "wip:")
+		isDraft := hasPrefix(title, "draft:") ||
+			hasPrefix(title, "[draft]") ||
+			hasPrefix(title, "(draft)")
 
 		if !isWIP && !isDraft {
 			title = "Draft: " + title

--- a/cmd/mr_create.go
+++ b/cmd/mr_create.go
@@ -218,6 +218,10 @@ func runMRCreate(cmd *cobra.Command, args []string) {
 		}
 	}
 
+	if title == "" {
+		log.Fatal("aborting MR due to empty MR msg")
+	}
+
 	linebreak, _ := cmd.Flags().GetBool("force-linebreak")
 	if linebreak {
 		body = textToMarkdown(body)
@@ -238,10 +242,6 @@ func runMRCreate(cmd *cobra.Command, args []string) {
 	removeSourceBranch, _ := cmd.Flags().GetBool("remove-source-branch")
 	squash, _ := cmd.Flags().GetBool("squash")
 	allowCollaboration, _ := cmd.Flags().GetBool("allow-collaboration")
-
-	if title == "" {
-		log.Fatal("aborting MR due to empty MR msg")
-	}
 
 	mrURL, err := lab.MRCreate(sourceProjectName, &gitlab.CreateMergeRequestOptions{
 		SourceBranch:       &branch,

--- a/cmd/util.go
+++ b/cmd/util.go
@@ -498,6 +498,14 @@ func determineSourceRemote(branch string) string {
 	return forkRemote
 }
 
+// Check of a case-insensitive prefix in a string
+func hasPrefix(str, prefix string) bool {
+	if len(str) < len(prefix) {
+		return false
+	}
+	return strings.EqualFold(str[0:len(prefix)], prefix)
+}
+
 // union returns all the unique elements in a and b
 func union(a, b []string) []string {
 	mb := map[string]bool{}

